functions/venue.js

```diff
@@ -462,20 +548,29 @@ exports.toggleDustStorm = functions.https.onCall(async (_data, context) => {
   }
 });
 
+// @debt this is almost a line for line duplicate of exports.updateVenue_v2, we should de-duplicate/DRY these up
 exports.updateVenue = functions.https.onCall(async (data, context) => {
   const venueId = data.id || getVenueId(data.name);
   checkAuth(context);
 
+  // @debt updateVenue_v2 uses checkUserIsAdminOrOwner rather than checkUserIsOwner. Should these be the same? Which is correct?
   await checkUserIsOwner(venueId, context.auth.token.user_id);
 
+  // @debt We should validate venueId conforms to our valid patterns before attempting to use it in a query
   const doc = await admin.firestore().collection("venues").doc(venueId).get();
 
+  // @debt this is exactly the same as in updateVenue_v2
   if (!doc || !doc.exists) {
     throw new HttpsError("not-found", `Venue ${venueId} not found`);
   }
 
+  // @debt this is exactly the same as in updateVenue_v2
   const updated = doc.data();
 
+  // note: This function will mutate the passed in updated object
+  createBaseUpdateVenueData(data, updated);
```

Comment:
IMO this should be returning the updated object instead of mutating it. Also, the second object is called `updated` but it's not actually updated yet.

Reply:
Yeah, I agree, but I couldn’t wrap my brain around it last night, and this is basically the equivalent pattern of how it was already implemented, just extracting the common parts into a shared function.

I didn’t want to have to make a deep clone of the existing object, and I didn’t want to have to build up the deeply nested structures and deal with potentially undefined fields and such.

I absolutely expect we’ll iterate and improve on this more in future, but I kind of want to wait till we have typescript in place on the backend to reduce the cognitive burden of trying to do so.

Given this is more or less equivalent to the existing pattern, and the importance of some of the fixes here, I’d like to get it out if there’s no ‘major issues/regressions’ with it.

Reply:
If you have some quick/efficient solutions to avoiding the mutation I’m happy to hear them. But otherwise I think we should call that a future iterations improvement.

Reply:
Maybe just rename them for now, anything in the form of source/destination, old/new  and maybe mark the intent for the future with @debt

Reply:
`@debt` was added in https://github.com/sparkletown/sparkle/pull/1485/commits/682627b2ba302bdf6cf5c58d0c4a8fc71fca84a4

Nuance of naming and 'not yet updated' is pre-existing from the code that was here before, so calling OOS.